### ISSUE TITLE
Add commit parsing of keyword "[conan promote]", invoking a copy

### ISSUE THREAD

**Issue by solvingj** [conan-io/conan-package-tools#266]
i would like to add parsing in Conan Package Tools for a special commit-message-keyword [conan promote].  this will invoke a conan copy from testing to stable branch, which is effectively a promotion that bypasses the rebuild.   we should get this in place before we go through and rebuild boost for 1.67 and 1.68.0. 

In summary, we very much like the git flow of making sure CI passes on the testing branch of the recipe repo's, and then merging into the stable branch.  However, with Boost, Qt, and larger packages, the rebuild caused by the merging is simply far too resource intensive, and strictly unnecessary.

**Comment by uilianries** [comment]
Some time ago I needed to promote my packages at my job but I wasn't smart enough to re-use CPT to do this, so I created my own project:

https://gitlab.com/uilianries/conan-promote

**Comment by solvingj** [comment]
yes, testing packages for boost are being uploaded. however, they all reference `@bincrafters/testing`, so probably this won't work because conan wont change references in conanfile without breaking or outdating the recipe, right?  it would have to update and and re-export.

**Comment by solvingj** [comment]
for boost, we're now using `python_requires` and adding other `boost` requirements dynamically in base class:  https://github.com/bincrafters/conan-boost_locale/blob/testing/1.67.0/conanfile.py
Some boost recipes still have non-boost requirements added in a separate function to work properly with a base class, like `icu` and `bzip`, etc. 

Yes, as you said: `conan install` / `conan copy` / `conan upload` instead of `conan create`. 

As for how I see it, the purpose of Conan Package Tools seems to be automating common Conan/CI tasks.  The purpose of the `conan copy` command seems to be to promote packages from `testing` to `stable`, a very desirable Conan/CI task.   It seems very logical and straightforward to implement this into CPT. 

However, with all that said, I doubt it's feasible because of the problem of requirements that I mentioned earlier.  It seems to be an unfortunate side effect of our strategy to build and test multiple recipes together under a `testing` channel, and then find/replace our way through promotion to `stable` channel.  I don't see a good testing/promotion workflow for something as unique as the boost packages.
